src/renderers/WebGLRenderer.js

```diff
@@ -1099,6 +1103,16 @@ function WebGLRenderer( parameters ) {
 
 		}
 
+		var tempHandler = _this.onBeforeRender;
+
+		if ( tempHandler ) {
+
+			_this._onBeforeRender = null;
```

Comment:
I would put this on your app code instead. We do similar things when using `object.onBeforeRender` already:

https://github.com/mrdoob/three.js/blob/dev/examples/js/objects/Reflector.js#L158-L162